### PR TITLE
Add the metadata controller chart.

### PR DESCRIPTION
**Pull request by blairboy362**
A chart for the metadata controller.

Solo: @blairboy362

### CODE REVIEW

File: proxy-node-chart/charts/metadata-controller/templates/customer-ca-crt-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ template "metadata-controller.fullname" . }}
+  labels:
+    app: {{ template "metadata-controller.name" . }}
+    chart: {{ template "metadata-controller.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  hsm_signing_cert.pem: {{ required "A valid .Values.customerCaCrt entry required!" .Values.customerCaCrt | b64enc | quote }}
```
Comment (vixus0) [inline proxy-node-chart/charts/metadata-controller/templates/customer-ca-crt-configmap.yaml]: You probably don't mean to use `customerCaCrt` here, right? The `hsm_signing_cert` is the cert attached to the private key in the HSM while the `customerCACrt` is used by the cloudhsm_client to access the HSM cluster.

File: proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}-cloudhsm-client
```
Comment (vixus0) [inline proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml]: As discussed the cloudhsm client should be a separate deployment/pod.

File: proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                key: HSM_PASSWORD
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+        volumeMounts:
+          name: hsm-signing-cert
+          mountPath: /etc/verify-proxy-node
+      volumes:
+        - name: customer-ca-crt
```
Comment (vixus0) [inline proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml]: As above, only the cloudhsm client deployment/pod needs the customer CA cert, not the metadata controller.

File: proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml (hunk 2 above)
Comment (chrisfarms) [reply proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml]: we decided to place cloudhsm client in sibling container afterall

File: proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml (hunk 3 above)
Comment (chrisfarms) [reply proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml]: we decided to place cloudhsm client in sibling container afterall - so this volume is required for the hsm client again

File: proxy-node-chart/charts/metadata-controller/templates/customer-ca-crt-configmap.yaml (hunk 1 above)
Comment (chrisfarms) [reply proxy-node-chart/charts/metadata-controller/templates/customer-ca-crt-configmap.yaml]: I ported the change to the incorrect configmap name + key

**Pull request merged**